worlds/timespinner/Options.py

```diff
@@ -1,611 +1,658 @@
+    }
+
+
+class UnchainedKeys(Toggle):
+    """Start with Twin Pyramid Key, which does not give free warp;
+    warp items for Past, Present, (and ??? with Enter Sandman) can be found."""
+    display_name = "Unchained Keys"
+
+
+class TrapChance(Range):
+    """Chance of traps in the item pool.
+    Traps will only replace filler items such as potions, vials and antidotes"""
+    display_name = "Trap Chance"
+    range_start = 0
+    range_end = 100
+    default = 10
+
+
+class Traps(OptionList):
+    """List of traps that may be in the item pool to find"""
+    display_name = "Traps Types"
+    valid_keys = {"Meteor Sparrow Trap", "Poison Trap", "Chaos Trap", "Neurotoxin Trap", "Bee Trap"}
+    default = ["Meteor Sparrow Trap", "Poison Trap", "Chaos Trap", "Neurotoxin Trap", "Bee Trap"]
+
+
+class PresentAccessWithWheelAndSpindle(Toggle):
+    """When inverted, allows using the refugee camp warp when both the Timespinner Wheel and Spindle is acquired."""
+    display_name = "Back to the future"
+
+
+@dataclass
+class TimespinnerOptions(PerGameCommonOptions, DeathLinkMixin):
+    start_with_jewelry_box: StartWithJewelryBox
+    downloadable_items: DownloadableItems
+    eye_spy: EyeSpy
+    start_with_meyef: StartWithMeyef
+    quick_seed: QuickSeed
+    specific_keycards: SpecificKeycards
+    inverted: Inverted
+    gyre_archives: GyreArchives
+    cantoran: Cantoran
+    lore_checks: LoreChecks
+    boss_rando: BossRando
+    damage_rando: DamageRando
+    damage_rando_overrides: DamageRandoOverrides
+    hp_cap: HpCap
+    level_cap: LevelCap
+    extra_earrings_xp: ExtraEarringsXP
+    boss_healing: BossHealing
+    shop_fill: ShopFill
+    shop_warp_shards: ShopWarpShards
+    shop_multiplier: ShopMultiplier
+    loot_pool: LootPool
+    drop_rate_category: DropRateCategory
+    fixed_drop_rate: FixedDropRate
+    loot_tier_distro: LootTierDistro
+    show_bestiary: ShowBestiary
+    show_drops: ShowDrops
+    enter_sandman: EnterSandman
+    dad_percent: DadPercent
+    rising_tides: RisingTides
+    rising_tides_overrides: RisingTidesOverrides
+    unchained_keys: UnchainedKeys
+    back_to_the_future: PresentAccessWithWheelAndSpindle
+    trap_chance: TrapChance
+    traps: Traps
+
+
+class HiddenDamageRandoOverrides(DamageRandoOverrides):
+    """Manual +/-/normal odds for an orb. Put 0 if you don't want a certain nerf or buff to be a possibility. Orbs that
+    you don't specify will roll with 1/1/1 as odds"""
+    visibility = Visibility.none
+
+
+class HiddenRisingTidesOverrides(RisingTidesOverrides):
+    """Odds for specific areas to be flooded or drained, only has effect when RisingTides is on.
+    Areas that are not specified will roll with the default 33% chance of getting flooded or drained"""
+    visibility = Visibility.none
+
+
+class HiddenTraps(Traps):
+    """List of traps that may be in the item pool to find"""
+    visibility = Visibility.none
+
+
+class HiddenDeathLink(DeathLink):

```

Comment:
What does this actually cover / allow for?

Reply:
Its used for the option `DeathLink`, that is the old casing instead of `death_link`
These old casing should be hidden as we dont want them to show up anywhere, but older yamls that still use the old casings will still work due to the `handle_backward_compatibility()`